java/code/src/com/redhat/rhn/domain/image/DeltaImageInfo.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2021 SUSE LLC
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package com.redhat.rhn.domain.image;
+
+import com.redhat.rhn.domain.BaseDomainHelper;
+import com.redhat.rhn.domain.server.Pillar;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.IdClass;
+import javax.persistence.JoinColumn;
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
+
+/**
+ * DeltaImageInfo
+ */
+@Entity
+@IdClass(DeltaImageInfoKey.class)
+@Table(name = "suseDeltaImageInfo")
+public class DeltaImageInfo extends BaseDomainHelper {
+
+    private ImageInfo sourceImageInfo;
+    private ImageInfo targetImageInfo;
+    private Pillar pillar;
```

Comment:
since we have the pillar data on both source and target images version, why do we need the pillar on the diff?
Is a utility data with also the compute delta on the pillar data?

Reply:
Yes, delta has additional data to the image pillars - url to sync delta from, delta file checksum hash and size.